core/src/main/java/org/bitcoinj/core/TransactionBroadcast.java

```diff
@@ -130,11 +134,30 @@ public Message onPreMessageReceived(Peer peer, Message m) {
         }
     };
 
+    /**
+     * Broadcast this transaction to the proper calculated number of peers. Returns a future that completes when the message
+     * has is "seen" by remote peers. The {@link Transaction} itself is the returned type/value for the future.
+     * <p>
+     * The broadcast process includes the following steps:
+     * <ol>
+     *     <li>Wait until enough {@link org.bitcoinj.core.Peer}s are connected.</li>
+     *     <li>Broadcast the transaction by a determined number of {@link org.bitcoinj.core.Peer}s</li>
+     *     <li>Wait for confirmation from a determined number of remote peers that they have received the broadcast</li>
+     *     <li>Mark {@link TransactionBroadcast#future()} ("sent future") as complete</li>
+     * </ol>
+     * <p>
+     * It should further be noted that "broadcast" in this class means that
+     * {@link org.bitcoinj.net.MessageWriteTarget#writeBytes} has completed successfully which means the message has
+     * been sent to the "OS network buffer" -- see {@link org.bitcoinj.net.MessageWriteTarget#writeBytes} or its implementation.
+     * <p>
+     * TODO: Should this method be moved into the PeerGroup?
```

Comment:
I think this TODO should not go into the JavaDoc.

Reply:
OK, I'll fix this.